### DISCUSSION TITLE
how to enable database query tracing with useSentry plugin

### DISCUSSION THREAD

**Discussion by alexanderniebuhr** [dotansimha/envelop#1448] [Q&A] [answered]
Hello everyone,

I hope to get some help. We currently use the [`useSentry`](https://www.envelop.dev/plugins/use-sentry) plugin to trace our GraphQL Yoga server.
According to the [Sentry documentation](https://docs.sentry.io/platforms/node/guides/connect/performance/database/auto-instrument/), the default node.js client `pg` should get automatically picked up and added to the traces. However, we can't see the detailed Postgres Traces through the plugin. 
It would be great to get an insight into the span of such resolvers with database queries.

Has anyone successfully configured it or can help to figure it out?

**Comment by kamilkisiela** [comment, marked as answer]
The `useSentry` acts on an already instantiated `Sentry` client, so I don't think it has anything to do with auto-instrumentation.
Could you share your setup and confirm that using pure sentry, without the envelop plugin works fine with `pg`?

**Reply by alexanderniebuhr** [reply]
@kamilkisiela I made a reproduction repository, which you can find at this location: https://github.com/alexanderniebuhr/envelop-sentry-postgres
_We use this setup but in a larger application that I can't share publicly._

As you can see, the spans of the sentry transaction are different. This is because, on the pure sentry way, it picks up the SQL query as a `db` operation and puts the query in the span' description.
However, on envelop one, it just spans the whole resolver. While this is great, I would like to have the entire resolver as a span and additional the `db` operation (so just the query part of the resolver) as a additional (child) span. Am I missing some configuration, or is this an unsupported edge case?

**Reply by kamilkisiela** [reply]
https://github.com/n1ru4l/envelop/pull/1463

**Reply by alexanderniebuhr** [reply]
Thanks for the quick fix. I can verify with the alpha release, that it works. 😍
